### PR TITLE
Update ContactDelete, Add ContactEdit, TaskEdit

### LINKED ISSUE #14: As a user, I can edit my contact's information
so that I can keep their information up to date.

### LINKED ISSUE #132: Block deleting/editing Person if associated to some Task
1. Handle `ContactDelete` and block it if there is an existing Task associated to the Person.
2. Handle `ContactEdit` and change the Task's Name associated to the Person.

A task should not be missing a person.

### LINKED ISSUE #142: As a user, I can edit my task's information
so that I can keep the information up to date.

### PR DESCRIPTION
**Pull request by seetohjinwei**
Fixes #14, fixes #132, fixes #142.

* Updated ContactDelete to block deletion of contact if there is still some task associated to that contact.
* Added ContactEdit, editing contact's name will propagate change to all associated tasks to that contact.
* Added TaskEdit

### CODE REVIEW

File: src/test/java/taskbook/logic/parser/contacts/ContactCategoryParserTest.java [hunk 1]
```diff
@@ -33,4 +34,11 @@ public void parseCommand_list() throws Exception {
         Command command = ContactCategoryParser.parseCommand(ContactListCommand.COMMAND_WORD, arguments);
         assertTrue(command instanceof ContactListCommand);
     }
+
+    @Test
+    public void parseCommand_edit() throws Exception {
+        String arguments = " i/1 n/Editing name";
```
Comment (dexter-sim) [inline src/test/java/taskbook/logic/parser/contacts/ContactCategoryParserTest.java:40]: Would it be a good idea to have this comment each time a string that has a white space at the start is used?

`// Note: the space at the start of the arguments is necessary due to ArgumentTokenizer behavior.`

**Review by dexter-sim — APPROVED** [review]
lgtm

File: src/test/java/taskbook/logic/parser/contacts/ContactCategoryParserTest.java (hunk 1 above)
Comment (seetohjinwei) [reply src/test/java/taskbook/logic/parser/contacts/ContactCategoryParserTest.java:40]: Personally, I think it's not necessary! I feel like we can include that in the devloper guide though! But if you insist, I can add it in too.

Comment (dexter-sim) [reply src/test/java/taskbook/logic/parser/contacts/ContactCategoryParserTest.java:40]: Agree. Having it in the developer guide is sufficient.

**Review by xiaobill8 — APPROVED** [review]
LGTM!

**Pull request merged**